.evergreen/run-serverless-tests.sh

```diff
@@ -12,10 +12,10 @@ if [ -z ${SERVERLESS_ATLAS_USER+omitted} ]; then echo "SERVERLESS_ATLAS_USER is
 if [ -z ${SERVERLESS_ATLAS_PASSWORD+omitted} ]; then echo "SERVERLESS_ATLAS_PASSWORD is unset" && exit 1; fi
 
 npx mocha --file test/tools/runner/index.js \
-  test/functional/crud_spec.test.js \
-  test/functional/retryable_reads.test.js \
-  test/functional/retryable_writes.test.js \
+  test/integration/crud/crud.spec.test.js \
+  test/integration/retryable-reads/retryable_reads.spec.test.js \
+  test/integration/retryable-writes/retryable_writes.spec.test.js \
   test/functional/sessions.test.js \
```

Comment:
Transaction and session tests both have tests that aren't the automated spec tests, so would end up getting separated upon moving, and since the serverless test spec requires us to run the prose tests, too (which aren't clearly identified in these combined test files), I think I'd rather do it in a separate PR from this one